pkg/util/validation/limits_test.go

```diff
@@ -110,6 +112,43 @@ func TestLimitsLoadingFromYaml(t *testing.T) {
 	assert.Equal(t, 100, l.MaxLabelNameLength, "from defaults")
 }
 
+func TestLimitsLoadingFromJson(t *testing.T) {
+	SetDefaultLimitsForYAMLUnmarshalling(Limits{
+		MaxLabelNameLength: 100,
+	})
+
+	inp := `{"ingestion_rate": 0.5}`
+
+	l := Limits{}
+	err := json.Unmarshal([]byte(inp), &l)
+	require.NoError(t, err)
+
+	assert.Equal(t, 0.5, l.IngestionRate, "from json")
+	assert.Equal(t, 100, l.MaxLabelNameLength, "from defaults")
+}
+
+func TestLimitsTagsYamlMatchJson(t *testing.T) {
```

Comment:
Great idea for a test

Reply:
Agree, that's pretty cool. Good job!

Reply:
This test will pass anyway if a file has only YAML and another field has only JSON. Have you considered checking that the yaml and json tag of each field exactly match (the content I mean)?

Reply:
Good point. I'll adjust the test to fix that :+1: